arelle/plugin/validate/UK/ValidateUK.py

```diff
@@ -646,6 +651,31 @@ def _evaluateProfLossOrCharityFundsFact(self, code: str) -> CodeResult:
             )
         return CodeResult()
 
+    def _evaluateGroupFacts(self) -> CodeResult:
+        """
+        Logs an error if BalanceSheetDate with GroupCompanyDataDimension and Consolidated must exist as non-nil and
+        equal BalanceSheetDate with GroupCompanyDataDimension and Consolidated
+        """
+        consolidatedFact = None
+        defaultFact = None
+        for balanceSheetDateFact in self._getFacts(CONCEPT_BALANCE_SHEET_DATE):
+            if self._checkValidFact(balanceSheetDateFact):
+                if not balanceSheetDateFact.context.qnameDims:
+                    defaultFact = balanceSheetDateFact
```

Comment:
Can there be more than one non-dimensional fact? Such as tagged for 2024 and 2024, or tagged with USD or EUR.

Reply:
Should be, but what happens if the do? Is a different error raised?

Reply:
If there are multiple, then the duplicate fact check should kick in. Because this is a date fact and with default dimensions any duplicates of this fact that are not consistent would be flagged.